apps/rush/src/actions/BaseAction.ts

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+import {
+  CommandLineAction,
+  ICommandLineActionOptions
+} from '@microsoft/ts-command-line';
+
+import {
+  RushConfiguration
+} from '@microsoft/rush-lib';
+
+import EventHooksManager from '../utilities/EventHooksManager';
+
+export abstract class BaseAction extends CommandLineAction {
```

Comment:
>BaseAction [](start = 22, length = 10)

The name "BaseAction" sounds even more general than CommandLineAction.  Maybe it should be a more specific name, like BaseRushAction?